packages/bitcore-lib/lib/crypto/signature.js

```diff
@@ -61,9 +64,18 @@ Signature.fromCompact = function(buf) {
 };
 
 Signature.fromDER = Signature.fromBuffer = function(buf, strict) {
-  var obj = Signature.parseDER(buf, strict);
   var sig = new Signature();
 
+  // Schnorr Signatures use 65 byte for in tx r [len] 32 , s [len] 32, nhashtype
+  // NOTE: this check is not very reliable. You should use .fromSchnorr directly if you know it's a schnorr sig.
+  if((buf.length === 64 || buf.length === 65) && buf[0] != 0x30) {
```

Comment:
Is there any way that we can have greater reliability here? One thing that jumped out at me is that a Signature has 'isSchnorr' property on it.

It seems like using that property here would be better. And if it needs to be initialized better, then you can do so in just one place, and there won't be these dangling problems.

Reply:
No. This is a `fromBuffer`/`fromDER` which is a defined encoded format. If all you have are a sequence of bytes, this is the best you can do to assert that it's a Schnorr sig or not. I could add a `isSchnorr` param to the function, but, like my comment says, if you know it's a Schnorr sig then just use .fromSchnorr()